src/features/RecipeEdit/RecipeAddController.tsx

```diff
@@ -6,10 +6,13 @@ import { useCreateRecipe } from "data/hooks/useCreateRecipe";
 import { useGetAllLabels } from "data/hooks/useGetAllLabels";
 import ClientId from "util/ClientId";
 import { useHistory } from "react-router-dom";
+import { ApolloError } from "@apollo/client";
+import { Alert } from "@mui/material";
 
 export const RecipeAddController = () => {
     const { data: labelList } = useGetAllLabels();
     const { createRecipe } = useCreateRecipe();
+    const [error, setError] = React.useState<ApolloError | null>(null);
```

Comment:
Wouldn't getting `error` from `useCreateRecipe` do basically the same thing, without needing to manually manage state and promises, and support "you tried again and it worked" for free?

Reply:
It sure would. That was silly of me. I shall change.